### DISCUSSION TITLE
Add onSuccess/onError to useWaitForTransactionReceipt

### DISCUSSION THREAD

**Discussion by ryanberckmans** [wevm/wagmi#4100] [Ideas] [answered]
onSuccess/onError used to exist on the old useWaitForTransactionReceipt and have since been removed.

Can you re-add them?

**Comment by tmm** [comment, marked as answer]
onSuccess/onError/onSettled won't be re-added. If you haven't already, you should read this blog post for more info on why there were removed https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose#react-query-v5

**Reply by ryanberckmans** [reply]
I hadn't seen that post, thanks.
